workflow/controller/operator.go

```diff
@@ -2977,6 +2978,33 @@ func (woc *wfOperationCtx) executeSuspend(nodeName string, templateScope string,
 	return node, nil
 }
 
+func (woc *wfOperationCtx) resolveInputFieldsForSuspendNode(node *wfv1.NodeStatus) {
+	if node.Inputs == nil || node.Inputs.Parameters == nil {
```

Comment:
minor - but if you range over `nil`, that is fine, so you don't need `node.Inputs.Parameters == nil `

Reply:
fixed